Formula/libgsm.rb

```diff
@@ -35,8 +41,12 @@ def install
     man3.mkpath
 
     # Dynamic library must be built first
-    system "make", "lib/libgsm.1.0.13.dylib",
-           "CC=#{ENV.cc}", "CCFLAGS=#{ENV.cflags}",
+    library = "libgsm.1.0.13.dylib"
+    on_linux { library = "libgsm.so" }
```

Comment:
The fix in linuxbrew-core does not version the shared library on Linux like we do on macOS.  Should we do this on Linux too?

Reply:
Probably, yes.

Reply:
I just noticed that this is already taken care of by the patch we are using (which is failing because of the Ruby error posted below).  However, the so version in the patch is out of date (1.0.18).  @iMichka I think we should move that patch from your repo to https://github.com/Homebrew/formula-patches so other maintainers can update it.

I also noticed that the dylib version for macOS is very outdated (1.0.13), and we should probably change the formula so that it fills in version automatically.  `libgsm` has no dependents so nothing else needs a revision bump if we do this.

Reply:
We should maybe put an `@@SOVERSION@@` placeholder in the patch that we `inreplace` in the formula so that the patch doesn't need to be manually updated.